core/charm/repository/sanitizer.go

```diff
@@ -28,25 +25,20 @@ func sanitizeCharmOrigin(received, requested corecharm.Origin) (corecharm.Origin
 		}
 	}
 
-	if result.Platform.OS == "all" {
-		result.Platform.OS = ""
-	}
-
-	if result.Platform.Series == "all" {
-		result.Platform.Series = ""
+	if result.Platform.Channel == "all" {
+		result.Platform.Channel = ""
 
-		if requested.Platform.Series != "all" {
-			result.Platform.Series = requested.Platform.Series
+		if requested.Platform.Channel != "all" {
+			result.Platform.Channel = requested.Platform.Channel
 		}
-	}
-
-	if result.Platform.Series != "" {
-		os, err := series.GetOSFromSeries(result.Platform.Series)
```

Comment:
why are we removing this added insurance step?

Reply:
We now check Channel instead. Os name and channel is the key tuple. Series is just derived from that and is only retained in 2.9 to send across the wire to older controllers. It will be dropped entirely in 3.0 when channel is used instead.